tests/codegen/vec-reserve-extend.rs

```diff
@@ -0,0 +1,14 @@
+// compile-flags: -O
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @should_reserve_once
+#[no_mangle]
+pub fn should_reserve_once(v: &mut Vec<u8>) {
+    // CHECK: tail call void @llvm.assume
+    v.try_reserve(3).unwrap();
+    // CHECK-NOT: call {{.*}}reserve
+    // CHECK-NOT: call {{.*}}do_reserve_and_handle
+    // CHECK-NOT: call {{.*}}__rust_alloc(
+    v.extend([1, 2, 3]);
```

Comment:
Another test with one or more calls to `push` instead of `extend` could be useful. If that works then this PR would also solve #105156.

Reply:
It won't solve the other issue, because `push` reserves capacity using a different non-inlined method, and needs another hint.